### ISSUE TITLE
Include category and/or tags in the SUMMARY?

### ISSUE THREAD

**Issue by aherranz** [dengste/org-caldav#14]
Hi, thanks for these great piece of software,

is there any way to include tags and categories in the summary of the exported entries?

Rationale: my org calendar entries (scheduled actions, meetings, deadlines, etc.) are structured in "tagged" and "categorized" "projects", a tipical headline is "Meeting". The entry in my google calendar is then "Meeting" but I don't know the project the meeting belongs.

**Comment by satran** [comment]
@aherranz more than 10 years later. I don't know whether you solved this or not but I think I have. I ran into the same trouble. For anyone else who is facing the same problem, here is how you do it. 

This needs to be set in the icalendar export settings. The variable to be configured is `org-icalendar-categories`. This is what I use:

```
(setq org-icalendar-categories '(all-tags local-tags))
```

This synchronises the tags using the category field of icalendar format.

**Comment by satran** [comment]
That sadly wasn't the case. It worked for some reason I cannot comprehend. But later it didn't. I started digging through the code and saw this line:

https://github.com/dengste/org-caldav/blob/44a6d463cee3c3be8acf7511db785ab55519b375/org-caldav.el#L1942

This resets the tags, regardless of if it was set. I don't understand the initial motivation. It is the one that is used to limit the items that needs to be synchronised by tags. But when synchronising back I'm not sure why this should be done. Commenting the line out solves this specific problem. 

@dengste thoughts?

**Comment by channel-42** [comment]
@satran I was able to get tag syncing working with your comments + by setting `(setq org-caldav-sync-changes-to-org 'all)` 👍🏼  Keep in mind that this overwrites some org specifics like logbooks etc., like stated in the vars docstring, which I can live with  🙂 

I haven't run into any issues so far, although I don't use the tag filtering/exclusion feature of org-caldav.

tl;dr workaround with el-patch:
```elisp
(use-package org-caldav
    :init/el-patch
    ;; NOTE: patch enables syncing of tags org <-> caldav-server
    (defun org-caldav-insert-org-entry--wrapup ()
      "Helper function to finish inserting an org entry or todo.
Sets the block's tags, and return its MD5."
      (org-back-to-heading)
      (el-patch-remove
        (org-caldav--org-set-tags-to org-caldav-select-tags))
      (md5 (buffer-substring-no-properties
	        (org-entry-beginning-position)
	        (org-entry-end-position))))
    :custom
    ;; insert your org-caldav config
    (org-caldav-sync-changes-to-org 'all)
    (org-icalendar-categories '(all-tags local-tags)))
```

**Comment by jackkamm** [comment]
The more recent comments about tag sync are unrelated to the original issue, which was a feature request to put tags into the title of the exported iCal event.

Tag sync is a more recent addition, and was only partially implemented for Todos (not events), and also broken. I've fixed the tag sync now and it works for both events and todos, in this commit: https://github.com/dengste/org-caldav/commit/53b9f2f67605409113803040de4a79cbc7561f2b

As for the original feature request (to have an option to insert tags into the title on iCal side), we won't implement it, as it wouldn't work with 2way sync.
